src/services/organizationService.ts

```diff
@@ -490,6 +495,104 @@ export async function registerOrganization(
   }
 }
 
+export function addDelegate(
+  ipaCode: string,
+  userEmail: string
+): TaskEither<
+  // tslint:disable-next-line:max-union-size
+  IResponseErrorInternal | IResponseErrorNotFound | IResponseErrorConflict,
+  IResponseSuccessRedirectToResource<Organization, Organization>
+> {
+  const genericErrorHandler = (error: unknown) => {
+    log.error("An error occurred adding the delegate. %s", error);
+    return ResponseErrorInternal("An error occurred adding the delegate.");
+  };
+  return tryCatch<
+    IResponseErrorInternal | IResponseErrorNotFound | IResponseErrorConflict,
+    OrganizationModel
+  >(
+    () =>
+      OrganizationModel.findOne({
+        where: { ipaCode }
+      }),
+    genericErrorHandler
+  )
+    .chain(
+      fromPredicate<IResponseErrorNotFound, OrganizationModel>(
```

Comment:
have you checked if these types specifications here can be inferred ? 
(in other words, if you omit them does the compiler complain ? if not, omit them : )